handsontable/src/core.js

```diff
@@ -2636,38 +2636,69 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
   };
 
   /**
-   * Allows altering the table structure by either inserting/removing rows or columns.
-   * This method works with an array data structure only.
+   * The `alter()` method lets you alter the grid's structure
+   * by adding or removing rows and columns at specified positions.
+   *
+   * ::: tip
+   * The `alter()` method works only when your [`data`](@/api/options.md#data)
+   * is an [array of arrays](@/guides/getting-started/binding-to-data.md#array-of-arrays).
+   * :::
+   *
+   * ```js
+   * // above row 10 (by visual index), insert 1 new row
+   * hot.alter('insert_row_above', 10);
+   * ```
+   *
+   *  | Action               | Description |
+   *  | -------------------- | ----------- |
+   *  | `'insert_row_above'` | Insert rows above a specified row. |
+   *  | `'insert_row_below'` | Insert rows below a specified row. |
+   *  | `'remove_row'`       | Remove specified rows. |
+   *  | `'insert_col_start'` | Insert columns before a specified column. |
+   *  | `'insert_col_end'`   | Insert columns after a specified column. |
+   *  | `'remove_col'`       | Remove specified columns. |
+   *  | `'insert_row'`       | (<b>Deprecated</b>) Insert rows above a specified row. |
+   *  | `'insert_col'`       | (<b>Deprecated</b>) Insert columns before a specified column. |
```

Comment:
I think that an important behavior, mentioned in https://github.com/handsontable/handsontable/pull/9955, should be clearly stated here:

```suggestion
   *  | `'insert_row_above'` | Insert rows above a specified row. If the `index` parameter is omitted, inserts above the first row. |
   *  | `'insert_row_below'` | Insert rows below a specified row. If the `index` parameter is omitted, inserts below the last row. |
   *  | `'remove_row'`       | Remove specified rows. |
   *  | `'insert_col_start'` | Insert columns before a specified column.  If the `index` parameter is omitted, inserts before the first column. |
   *  | `'insert_col_end'`   | Insert columns after a specified column.  If the `index` parameter is omitted, inserts after the last column. |
   *  | `'remove_col'`       | Remove specified columns. |
   *  | `'insert_row'`       | (<b>Deprecated</b>) Insert rows above a specified row.  If the `index` parameter is omitted, inserts below the last row. |
   *  | `'insert_col'`       | (<b>Deprecated</b>) Insert columns before a specified column.  If the `index` parameter is omitted, inserts after the last column. |
```

Reply:
@warpech – ok, I'd add it as another column to the table then:  [1f42cf2](https://github.com/handsontable/handsontable/pull/10046/commits/1f42cf2029d8833a96fa9d0182412356154a7be1) (I also marked `index` as optional, and added info about RTL at the bottom)